### PR TITLE
Ilia branch

### PR DESCRIPTION
**Pull request by IliaStepanov**
DB init from resource file and console DB view.

### CODE REVIEW

**Comment by vakhonin** [comment]
Веточку стараемся называть в соответсвии с задачей. Например, если стоит задача создания и подключения скрипта для инициализации БД, называем так: add-init-db, например так. Кратко, без подробностей. В описании пул реквеста можно распространить эту информацию.

File: src/main/java/com/epam/lowcost/DAO/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/DAO/.gitignore]: Is gitignore in DAO ? why ?

File: src/main/java/com/epam/lowcost/db/DBinitialize.java [hunk 2]
```diff
@@ -3,77 +3,72 @@
...
-        stm.executeUpdate(sql);
-        System.out.println("Inserted records into the table...");
+        File fl = new File(this.getClass().getResource("/createTable").getFile());
+        try (BufferedReader br = Files.newBufferedReader(fl.toPath())) {
 
+            String sql = br.readLine();
+            while (sql != null) {
+                stm.addBatch(sql);
+                sql = br.readLine();
+            }
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/db/DBinitialize.java]: https://www.mkyong.com/java8/java-8-stream-read-a-file-line-by-line/
You can try to use streams

File: src/main/java/com/epam/lowcost/DAO/.gitignore (hunk 1 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/DAO/.gitignore]: Чтобы пустая папка добавилась в репозиторий, как только появится любой файл, гитигнор можно будет убрать.

File: src/main/resources/createTable [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+CREATE TABLE REGISTRATION(id INTEGER not NULL,first VARCHAR(255),last VARCHAR(255),age INTEGER,PRIMARY KEY (id));
+--populating table
+INSERT INTO Registration VALUES (100, 'Zara', 'Ali', 18);
+INSERT INTO Registration VALUES (101, 'Mahnaz', 'Fatma', 25);
+INSERT INTO Registration VALUES (102, 'Zaid', 'Khan', 30);
+INSERT INTO Registration VALUES (103, 'Sumit', 'Mittal', 28);
+
+
+--creating table
+CREATE TABLE EMPLOYEES(id INTEGER not NULL,first VARCHAR(255),last VARCHAR(255),age INTEGER,PRIMARY KEY (id));
```
Comment (vakhonin) [inline src/main/resources/createTable]: Strange titles. What does it mean EMPLOYEE? REGISTRATION ?

File: src/main/java/com/epam/lowcost/db/DBinitialize.java [hunk 4]
```diff
@@ -3,77 +3,72 @@
...
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        stm.executeBatch();
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/db/DBinitialize.java]: I didn't find where you use close connection and statement.
```
Connection conn = null;
   Statement stmt = null;
   try{
      conn = DriverManager.getConnection(DB_URL, USER, PASS);
      stmt = conn.createStatement();
      String sql = "CREATE DATABASE STUDENTS";
      stmt.executeUpdate(sql);
   }catch(SQLException se){
      //Handle errors for JDBC
      se.printStackTrace();
   }catch(Exception e){
      //Handle errors for Class.forName
      e.printStackTrace();
   }finally{
      //finally block used to close resources
      try{
         if(stmt!=null)
            stmt.close();
      }catch(SQLException se2){
      }// nothing we can do
      try{
         if(conn!=null)
            conn.close();
      }catch(SQLException se){
         se.printStackTrace();
      }//end finally try
   }//end try
```
Something like this

File: src/main/resources/createTable (hunk 3 above)
Comment (IliaStepanov) [reply src/main/resources/createTable]: This is a sample data, while we do not have actual models. Could be changed to actual data like(USER, FLIGHT etc.) right now or after we add models to the project.

File: src/main/java/com/epam/lowcost/DAO/.gitignore (hunk 1 above)
Comment (vakhonin) [reply src/main/java/com/epam/lowcost/DAO/.gitignore]: It is not necessary. If we don't have files - ok, then in another task you will create file and we will see package.

File: src/main/resources/createTable (hunk 3 above)
Comment (vakhonin) [reply src/main/resources/createTable]: I can review DB scheme and then you add correct entities, ok ?

File: src/main/java/com/epam/lowcost/DAO/.gitignore (hunk 1 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/DAO/.gitignore]: Ok, I'll delete this .gitignore files.

File: src/main/resources/createTable (hunk 3 above)
Comment (IliaStepanov) [reply src/main/resources/createTable]: Will change it to correct entities right after DB scheme will be accepted, should I add all entities from DB scheme?

File: src/main/java/com/epam/lowcost/db/DBinitialize.java (hunk 2 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/db/DBinitialize.java]: Changed initiated method to stream realization.

File: src/main/java/com/epam/lowcost/db/DBinitialize.java (hunk 4 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/db/DBinitialize.java]: Changed DBinitialize constructor to close Connection and Statement.

File: src/main/java/com/epam/lowcost/DAO/.gitignore (hunk 1 above)
Comment (vakhonin) [reply src/main/java/com/epam/lowcost/DAO/.gitignore]: when you remove these files a will resolve this conversation.

Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/DAO/.gitignore]: I've deleted both files.

File: src/main/java/com/epam/lowcost/db/DBinitialize.java [hunk 5]
```diff
@@ -3,77 +3,85 @@
...
-                " first VARCHAR(255), " +
-                " last VARCHAR(255), " +
-                " age INTEGER, " +
-                " PRIMARY KEY ( id ))";
-
-        stm.executeUpdate(sql);
-
+        try {
+            conn = basicDataSource.getConnection();
+            stm = conn.createStatement();
```
[the remark this answers is not in the document]
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/db/DBinitialize.java]: Moved Connection and Statement  open\close  from constructor to method. Removed printAllDb method.

File: src/main/resources/createTable (hunk 3 above)
Comment (IliaStepanov) [reply src/main/resources/createTable]: Moved Connection and Statement  open\close  from constructor to method. Removed printAllDb method. As EugeneGrim requested.

Comment (vakhonin) [reply src/main/resources/createTable]: I can propose to create new task for creation correct data. If @EugeneGrim agree with this, he can resolve this discussion and merge this branch.

File: src/main/java/com/epam/lowcost/db/DBinitialize.java [hunk 6]
```diff
@@ -3,77 +3,85 @@
...
+                    for (int i = 1; i < columns; i++) {
+                        System.out.print(" " + rs.getMetaData().getColumnName(i) + "=" + rs.getObject(i));
+                    }
+                    System.out.println();
+                }
+                System.out.println();
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
```
[the remark this answers is not in the document]
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/db/DBinitialize.java]: Removed  printAllDb method completly

**Pull request merged**